### PR TITLE
bazel: update rules_rust

### LINKED ISSUE #21435: Newer release available `rules_rust`: 0.5.0 (current: 0.3.1)
Package Name: rules_rust@0.3.1
Current Version: 0.3.1@2022-04-26
Available Version: 0.5.0@2022-05-24
Upstream releases: https://github.com/bazelbuild/rules_rust/releases

### PR DESCRIPTION
**Pull request by keith**
Signed-off-by: Keith Smiley <keithbsmiley@gmail.com>

Fix #21435

### CODE REVIEW

**Comment by keith** [comment]
https://github.com/bazelbuild/rules_rust/issues/1330

**Comment by htuch** [comment]
LGTM when CI passes

**Comment by phlax** [comment]
5.0 is out https://github.com/envoyproxy/envoy/issues/21435

**Comment by keith** [comment]
I threw that in here, but it still doesn't fix our issue, there's an open pr for it over there

**Comment by moderation** [comment]
@keith is it worth a try with `0.6.0`?

https://github.com/bazelbuild/rules_rust/releases/tag/0.6.0

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -1125,12 +1125,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Bazel rust rules",
         project_desc = "Bazel rust rules (used by Wasm)",
         project_url = "https://github.com/bazelbuild/rules_rust",
-        version = "0.3.1",
-        sha256 = "e074f1e203607c5fcd549929d956170346f8807d2bbaeb98b2ed213c37e0870f",
+        version = "0.5.0",
+        sha256 = "73580f341f251f2fc633b73cdf74910f4da64d06a44c063cbf5c01b1de753ec1",
```
Comment (moderation) [inline bazel/repository_locations.bzl:1129]:
```python
        version = "0.6.0",
        sha256 = "872b04538ca20dad94791c348623f079ba93daf274c1d57ae6bfe0930ec77f0d",
```

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -1125,12 +1125,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
         urls = ["https://github.com/bazelbuild/rules_rust/releases/download/{version}/rules_rust-v{version}.tar.gz"],
         use_category = ["dataplane_ext"],
         extensions = ["envoy.wasm.runtime.wasmtime"],
-        release_date = "2022-04-26",
+        release_date = "2022-05-24",
```
Comment (moderation) [inline bazel/repository_locations.bzl:1133]: `        release_date = "2022-06-14",`

**Pull request closed without merging**